core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withEditorToolbar.ts

```diff
@@ -0,0 +1,203 @@
+ *
+ * @export
+ * @return {*}
+ */
+export function withEditorToolbar() {
+    return signalStoreFeature(
+        {
+            state: type<UVEState>()
+        },
+        withState<EditorToolbarState>(initialState),
+        withComputed((store) => ({
+            $toolbarProps: computed<ToolbarProps>(() => {
+                const params = store.$params();
+                const url = sanitizeURL(params.url);
+
+                const pageAPIQueryParams = createPageApiUrlWithQueryParams(url, params);
+                const pageAPIResponse = store.$pageAPIResponse();
+                const experiment = store.$experiment?.();
+
+                const pageAPI = `/api/v1/page/${
+                    store.$isTraditionalPage() ? 'render' : 'json'
+                }/${pageAPIQueryParams}`;
+
+                return {
+                    deviceSelector: {
+                        apiLink: `${params.clientHost ?? window.location.origin}${pageAPI}`,
+                        hideSocialMedia: !store.$isTraditionalPage()
+                    },
+                    urlContentMap: store.$isEditState() ? pageAPIResponse.urlContentMap : null,
+                    bookmarksUrl: createFavoritePagesURL({
+                        languageId: Number(params.language_id),
+                        pageURI: url,
+                        siteId: pageAPIResponse.site.identifier
+                    }),
+                    copyUrlButton: {
+                        pureURL: createPureURL(params)
+                    },
+                    apiLinkButton: {
+                        apiURL: `${window.location.origin}${pageAPI}`
+                    },
+                    experimentBadge:
+                        experiment?.status === DotExperimentStatus.RUNNING
+                            ? {
+                                  runningExperiment: experiment
+                              }
+                            : null,
+                    languageSelector: {
+                        currentLanguage: pageAPIResponse.viewAs.language
+                    },
+                    personaSelector: {
+                        pageId: pageAPIResponse.page.identifier,
+                        value: pageAPIResponse.viewAs.persona ?? DEFAULT_PERSONA
+                    },
+                    workflowActions: store.$canEditPage()
+                        ? {
+                              inode: pageAPIResponse.page.inode
+                          }
+                        : null,
+                    unlockButton:
+                        pageAPIResponse.page.locked && pageAPIResponse.page.canLock
+                            ? {
+                                  inode: pageAPIResponse.page.inode,
+                                  loading: store.$status() === UVE_STATUS.LOADING
+                              }
+                            : null,
+                    showInfoDisplay:
```

Comment:
Why all this nesting?

Reply:
Removed unnecesary nesting and code cleaned